lib/api/2.0/nodes.js

```diff
@@ -61,24 +62,45 @@ var nodesGetPollersById = controller(function(req) {
 });
 
 var nodesGetWorkflowById = controller(function(req) {
-    return nodes.getNodeWorkflowById(req.swagger.params.identifier.value);
+    return workflowApiService.getWorkflowsByNodeId(req.swagger.params.identifier.value,
```

Comment:
@benbp I'm not sure where the requirement for Cassandra comes from, but it's something that @zyoung51 has been working on for a while.  One possible way to address this particular problem is to add an `activeOnNode` property to workflows.  The value associated with this property would either be the node ID on which the workflow is active or `null` if the workflow is either inactive or detached from a node.  It would need to be updated atomically with the `_status` property on transitions between active and inactive states.  Then the active query becomes `{activeOnNode: nodeId}`. The drawback is that it becomes more difficult to list inactive nodes.  We could do it by taking the difference of the set of workflows active on a node and the complete set of workflows associated with a node but this has the potential for concurrency problems since a node could become active between the two queries.